trieste/models/keras/models.py

```diff
@@ -480,9 +481,18 @@ def optimize_encoded(self, dataset: Dataset) -> tf_keras.callbacks.History:
             fit_args["epochs"] = fit_args["epochs"] + self._absolute_epochs
 
         x, y = self.prepare_dataset(dataset)
+
+        if "validation_split" in fit_args:
+            (x, y), (x_val, y_val) = train_validation_split(
```

Comment:
`train_validation_split` is what keras uses internally

Reply:
the issue with validation data using output of `prepare_dataset` is that its typically bootstrapped since its needed for uncertainty quantification - thats not great for creating validation data, ideally we would separate out validation set and then create x,y that are bootstrapped for training - I address this in #899, i.e. modify methods and provided some utils to make it easier

Reply:
I've rebased this onto [#899](https://github.com/secondmind-labs/trieste/pull/899) and have updated it to form the validation split using `split_dataset_randomly` before potentially bootstrapping the data with `prepare_dataset`.